linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/executor/SparkSqlExecutor.scala

```diff
@@ -47,6 +44,35 @@ class SparkSqlExecutor(sparkEngineSession: SparkEngineSession, id: Long)
 
   override protected def getKind: Kind = SparkSQL()
 
+  // Only used in the scenario of direct pushing, dataFrame won't be fetched at a time,
+  // It will cache the lazy dataFrame in memory and return the result when client .
+  private def submitResultSetIterator(taskId: String, df: DataFrame): Unit = {
+    if (!DirectPushCache.isTaskCached(taskId)) {
+      DirectPushCache.submitExecuteResult(taskId, df)
+    } else {
+      logger.error(s"Task $taskId already exists in resultSet cache.")
+    }
+  }
+
+  override def isFetchMethodOfDirectPush(taskId: String): Boolean = {
+    DirectPushCache.isTaskCached(taskId)
+  }
+
+  // This method is not idempotent. After fetching a result set of size fetchSize each time, the corresponding results will be removed from the cache.
+  override def fetchMoreResultSet(taskId: String, fetchSize: Int): FetchResultResponse = {
+    val dataFrameResponse = DirectPushCache.fetchResultSetOfDataFrame(taskId, fetchSize)
+    if (dataFrameResponse.dataFrame != null) {
+      if (!dataFrameResponse.hasMoreData) {
+        succeedTasks.increase()
+      }
+      FetchResultResponse(
+        hasMoreData = dataFrameResponse.hasMoreData,
+        ArrowUtils.toArrow(dataFrameResponse.dataFrame)
+      )
+    }
```

Comment:
Else is missing here

Reply:
done. After removing the  job-way to fetch data. This method isn't needed anymore.